modules/stochastic_tools/src/surrogates/PolynomialChaos.C

```diff
@@ -78,7 +111,7 @@ PolynomialChaos::execute()
     // Loop over coefficients
     for (std::size_t i = 0; i < _ncoeff; ++i)
     {
-      Real val = _values[p - _sampler->getLocalRowBegin()];
+      Real val = (*_values_ptr)[p - _sampler->getLocalRowBegin()];
```

Comment:
I think this causing the bug. If you switch to below, it should work
```suggestion
      Real val = (*_values_ptr)[p];
```

Reply:
I will look into that. What doesn't make sense is that the current version, which is also working has this.

https://github.com/idaholab/moose/blob/72914c17e0622b943c2df646e6bb2ee69824bd1d/modules/stochastic_tools/src/userobject/PolynomialChaos.C#L81

Reply:
I would expect that this index would need to change based how how the _values data setup in parallel: replicated (don't need to substract) and distributed (do need to subtract). I was playing around with StochasticResults and making sure it was properly replicated, which it wasn't. But, that wasn't the original PR.

Reply:
Correct, we need to shift the values if distributed. I'm not sure the best way to access the parallel type of the VPP though.

Reply:
There is no method now, I will add it in along with a test that uses distributed data. Thanks again for helping me figure this one out.

Reply:
No problem. Glad I could help. Let me know if you run into any more problems.